File: frontend/src/logic/TicTacToeGameFactory.js
import { TicTacToeGame } from './TicTacToeGame.js';

export const GameType = {
    STANDARD: 'STANDARD',
    TIME_BATTLE : 'TIME_BATTLE'
};

export class TicTacToeGameFactory {
    static createGame(gt, boardSize) {
        if (gt === GameType.STANDARD) {
            return new TicTacToeGame(boardSize);
        }
        return null;
    }
}

File: TicTacToeGameFactory.cpp
enum GameType {
    STANDARD
};

class TicTacToeGameFactory {
public:
    static TicTacToeGame* createGame(GameType gt, int boardSize) {
        if(GameType::STANDARD == gt) {
            return new TicTacToeGame(boardSize);
        }
        return nullptr;
    }
};